File: ruby/lib/exercise1/symmetric.rb
module Symmetric

  def self.encrypt(key, message)
    box = RbNaCl::SimpleBox.from_secret_key(key)
    box.box(message)
  end

  def self.decrypt(key, ciphertext)
    box = RbNaCl::SimpleBox.from_secret_key(key)
    box.open(ciphertext)
  end

end

File: java/src/main/java/workshop/encryption/exercise1/Symmetric.java
package workshop.encryption.exercise1;

import helpers.HashedKey;
import helpers.RandomNonce;

import org.abstractj.kalium.crypto.SecretBox;

public class Symmetric {

    public static byte[] encrypt(HashedKey key, RandomNonce nonce, byte[] message) {
        SecretBox box = new SecretBox(key.getBytes());
        return box.encrypt(nonce.getBytes(), message);
    }

    public static byte[] decrypt(HashedKey key, RandomNonce nonce, byte[] encryptedMessage) {

        /**
         * TODO: implement this method
         * - take a look at the implementation of the #encrypt method
         * - use SecretBox#decrypt
         *
         */

        return null;
    }
}